scripts/kconfig/kconfigfunctions.py

```diff
@@ -0,0 +1,78 @@
+#
+# Copyright (c) 2018-2019 Linaro
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import os
+import re
+
+# Types we support
+# 'string', 'int', 'hex', 'bool'
+conf_file = os.path.join(os.environ['PROJECT_BINARY_DIR'], "include/generated/generated_dts_board.conf")
+
+dt_defines = {}
+if os.path.isfile(conf_file):
+    for line in open(conf_file, 'r', encoding='utf-8'):
+        if re.search('=', line):
+            c = line.split('=')
+            dt_defines[c[0]] = c[1].strip()
+
+def _dt_units_to_scale(unit):
+    if not unit:
+        return 0
+    if unit == 'k' or unit == 'K':
+        return 10
+    if unit == 'm' or unit == 'M':
+        return 20
+    if unit == 'g' or unit == 'G':
+        return 30
+
+def dt_int_val(kconf, name, arg, unit=None):
+    """
+    This function looks up 'name' in the DTS generated "conf" style database
+    and if its found it will return the value as an decimal integer.  The
+    function will scale the value based on 'unit':
+        'k' or 'K'  scale by 1024 (1 << 10)
+        'm' or 'M'  scale by 1,048,576 (1 << 20)
+        'g' or 'G'  scale by 1,073,741,824 (1 << 30)
+    """
+    try:
+        d = dt_defines[arg]
+        if d.startswith('0x') or d.startswith('0x'):
+            d = int(d, 16)
+        else:
+            d = int(d)
+        d = d >> _dt_units_to_scale(unit)
+
+    except KeyError:
+        d = 0
```

Comment:
Another option would be to error out in some way here I guess:

```python
sys.exit("{}:{}: unknown unit '{}'".format(kconf._filename, kconf._linenr))
```

That's using some internal variables to get the location, but it should be pretty safe.

Reply:
The sys.exit() might not work, is I think we need a default value for the cases in which the key doesn't exist.

For example if we build for posix since it doesn't have dt we need to default value and not error out in this case.

Reply:
Ah, didn't notice that the `try` also surrounds the `d = dt_defines[arg]`.

Maybe the code could do this instead (with no `try`), to make it a bit more explicit:

```python
if name not in dt_defines:
    return "0"

...
```

One thing that often confuses me in the `scripts/dts` code is that there are a lot of  `try` blocks around a bunch of code where it's not obvious what it means when you catch the `KeyError`, etc.

Reply:
That's from my lack of know going python style ;)

Reply:
Changed to do the `if` check and removed the try block.